0036-spsp-pull-payments/0036-spsp-pull-payments.md

```diff
@@ -0,0 +1,146 @@
+      * `sendMax` SHOULD be derived from the Pull Payment Agreement parameters, i.e., `sendMax` SHOULD be equal to `pull.balance.available`, converted to the SPSP Server's operating asset, taking exchange rate fluctuations into account.
+  2. The SPSP Client will adjust their `receiveMax` to reflect the amount they're willing to receive.
+      * `receiveMax` MAY be `Infinity`.
+  3. The SPSP Client's and Server's [STREAM Modules](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Definitions) will move as much value as possible while staying inside these bounds.
+  4. If the SPSP Server reaches their `sendMax`, they end the stream and the connection. If the SPSP Client reaches their `receiveMax`, they will end the stream and the connection.
+
+The STREAM parameters - `sendMax` and `receiveMax` - are defined in [STREAM's frame encoding](../0029-stream/0029-stream.md#53-frames).
+
+## Specification
+
+### Query (`GET <SPSP Endpoint>`)
+
+The SPSP Client queries the SPSP Endpoint to get information about the SPSP Server:
+
+#### Request
+
+``` http
+GET /0f09dc92-84ad-401b-a7c9-441bc6173f4e HTTP/1.1
+Host: alice.com
+Accept: application/spsp4+json
+```
+
+#### Response
+
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/spsp4+json
+
+{
+  "destination_account": "alice.ilpdemo.red.0f09dc92-84ad-401b-a7c9-441bc6173f4e",
+  "shared_secret": "k5nubgM6zpb88NPGVnI/tVjRdgpUh+JvMueRFEMvPcY=",
+  "pull": {
+    "balance": {
+      "total": "5000",
+      "interval": "1000",
+      "available": "1000"
+    },
+    "cycle": 3,
+    "agreement": {
+      "amount": "2000",
+      "start": "2019-01-01T08:00Z",
+      "expiry": "2021-01-02T00:00Z", 
+      "interval": "P0Y1M0DT0H0M0S",
+      "cycles": 12,
+      "cap": false,
+      "asset": {
+        "code": "USD",
+        "scale": 2
+      }
+    }
+  }
+}
+``` 
+
+##### Response Body
+
+The response body is a JSON object that includes basic account details necessary for setting up a STREAM connection as well as optional parameters for displaying purposes. The fields are defined in the following: 
+
+| Field | Type | Description |
+|---|---|---|
+| `destination_account` | [ILP Address](../0015-ilp-addresses/0015-ilp-addresses.md) | (see [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Response-Body)) |
+| `shared_secret` | 32 bytes, [base64 encoded](https://en.wikipedia.org/wiki/Base64) (including padding) | (see [Simple Payment Setup Protocol](../0009-simple-payment-setup-protocol/0009-simple-payment-setup-protocol.md#Response-Body)) |
+| `pull`  | Object | _(OPTIONAL)_ Details of this Pull Payment Pointer. |
+| `pull.balance`  | Object | _(OPTIONAL)_ Details of this Pull Payment Pointer's balance. |
+| `pull.balance.total` | Integer String | _(OPTIONAL)_ Total amount, denoted in `pull.agreement.asset.code`, which has been pulled by the client since `pull.agreement.start` to date. It is the sum of all outgoing chunks. |
```

Comment:
```suggestion
| `pull.balance.total` | Integer String | _(OPTIONAL)_ Total amount, denoted in `pull.agreement.asset`, which has been pulled by the client since `pull.agreement.start` to date. It is the sum of all outgoing chunks. |
```
or some other way of denoting both the code AND scale.
Ditto for

- `pull.balance.interval`
- `pull.balance.available`
- `pull.agreement.amount`

Reply:
Basically, you want less objects and more individual fields with precise names? I'm open for suggestions.